src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala

```diff
@@ -300,13 +300,16 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
       val assumedSourceRoot  = fixPath(settings.sourcepath.value) stripSuffix "/"
 
       if (!settings.docsourceurl.isDefault)
-        inSource map { case (file, _) =>
-          val filePath = fixPath(file.path).replaceFirst("^" + assumedSourceRoot, "").stripSuffix(".scala")
+        inSource map { case (file, line) =>
+          val filePathExt = fixPath(file.path).replaceFirst("^" + assumedSourceRoot, "")
+          val filePath = filePathExt.stripSuffix(".scala")
```

Comment:
shouldn't this be something like `filePathExt.stripSuffix(".scala").stripSuffix(".java")` to fix https://github.com/scala/bug/issues/5388?

Reply:
No, the existing use is as shown in https://github.com/scala/scala/pull/7532/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceL164 where the `.scala` is added by the URL template. Removing even the `.java` doesn't help and would make the current fix https://github.com/scala/bug/issues/5388#issuecomment-292401894 via post-processing impossible.

Reply:
I see what you mean, the documentation for `FILE_PATH` is still wrong for non-Scala files.  
I think it is a bad idea to change the meaning of `FILE_PATH`.
What about removing `FILE_PATH` from the documentation, but leaving it in the code for backwards-compatibility?

Reply:
I think the current `FILE_PATH` is buggy and we should just fix it. It would stay the same for scala files, which is the majority. Yes, the workaround would break, but I doubt a lot of people are doing this.

Reply:
I fixed `FILE_PATH` to never include the extension, added `FILE_EXT`, and left `FILE_PATH_EXT` to get the whole thing.

Reply:
Thanks, I was stubborn enough :-)